backend/routes/profiles.py

```
@@ -134,6 +134,26 @@ async def update_profile(
+    """Duplicate a voice profile, including its samples, avatar and settings.
+
+    Unlike an export/import round-trip this preserves personality, effects
+    chain, default engine and preset/designed fields, and works for profiles
+    with no samples.  Defaults the new name to "<name> (copy)".
+    """
+    try:
+        return await profiles.duplicate_profile(
+            profile_id, db, name=data.name if data else None
+        )
+    except ValueError as e:
+        raise HTTPException(status_code=404, detail=str(e))
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)